packages/manager/modules/exchange/src/account/mfa/account-mfa.controller.js

```diff
@@ -0,0 +1,140 @@
+    Exchange,
+    exchangeAccount,
+    messaging,
+    navigation,
+  ) {
+    this.$q = $q;
+    this.$scope = $scope;
+    this.$state = $state;
+    this.$translate = $translate;
+    this.exchangeAccount = exchangeAccount;
+    this.isLoading = false;
+    this.messaging = messaging;
+    this.navigation = navigation;
+    this.exchangeService = Exchange.value;
+    this.Exchange = Exchange;
+  }
+
+  $onInit() {
+    this.account = this.navigation.currentActionData.account;
+    this.action = this.navigation.currentActionData.action;
+    this.isLoading = true;
+    this.Exchange.getExchangeServer(
+      this.exchangeService.organization,
+      this.exchangeService.domain,
+    )
+      .then((server) => {
+        this.server = server;
+      })
+      .catch((error) => {
+        this.messaging.writeError(
+          this.$translate.instant('exchange_mfa_error', {
+            error: error.message,
+          }),
+        );
+
+        this.$scope.resetAction();
+      })
+      .finally(() => {
+        this.isLoading = false;
+      });
+  }
+
+  create() {
+    return this.exchangeAccount.createMfa(
+      this.exchangeService.domain,
+      this.account.primaryEmailAddress,
+    );
+  }
+
+  delete() {
+    return this.exchangeAccount.deleteMfa(
+      this.exchangeService.domain,
+      this.account.primaryEmailAddress,
+    );
+  }
+
+  disable() {
+    return this.exchangeAccount.disableMfa(
+      this.exchangeService.domain,
+      this.account.primaryEmailAddress,
+      this.disablePeriod,
+    );
+  }
+
+  enable() {
+    return this.exchangeAccount.enableMfa(
+      this.exchangeService.domain,
+      this.account.primaryEmailAddress,
+    );
+  }
+
+  reset() {
+    return this.exchangeAccount.resetMfa(
+      this.exchangeService.domain,
+      this.account.primaryEmailAddress,
+    );
+  }
+
+  submit() {
+    this.isLoading = true;
+
+    let accountMfaPromise = null;
+    let messages = {};
+    switch (this.action) {
+      case 'DELETE':
+        accountMfaPromise = this.delete();
+        messages = {
+          success: 'exchange_delete_mfa_success',
+          error: 'exchange_delete_mfa_error',
+        };
+        break;
+      case 'DISABLE':
+        accountMfaPromise = this.disable();
+        messages = {
+          success: 'exchange_disable_mfa_success',
+          error: 'exchange_disable_mfa_error',
+        };
+        break;
+      case 'ENABLE':
+        accountMfaPromise =
+          this.account.mfa.status === 'NOT_CREATED'
+            ? this.create()
+            : this.enable();
+        messages = {
+          success: 'exchange_enable_mfa_success',
+          error: 'exchange_enable_mfa_error',
+        };
+        break;
+      case 'RESET':
+        accountMfaPromise = this.reset();
+        messages = {
+          success: 'exchange_reset_mfa_success',
+          error: 'exchange_reset_mfa_error',
+        };
+        break;
+      default:
+        this.$scope.resetAction();
+    }
```

Comment:
Maybe we could use something like 
```suggestion
    accountMfaPromise = this[this.action.toLowerCase]() 
    messages = {
    success: `exchange_{this.action.toLowerCase}_mfa_success`
    ... 
```

Reply:
I was thinking of adapting another OOP pattern which is reflection 😄 
That way we don't have to add another `case` to add another behaviour (though we have to add a function)

Reply:
I know @marie-j, but I think it's possible it could do more harm than good

This articles sums up my opinion about this: https://overreacted.io/goodbye-clean-code/

Code duplication sometimes increases legibility and is easier to maintain than having the wrong abstraction